src/store/reducers/articlesReducer.ts

```diff
@@ -0,0 +1,36 @@
+import { createAsyncThunk, createSlice, PayloadAction } from "@reduxjs/toolkit";
+import axios from "axios";
+import { APIresponse } from "../../Utils/Types";
+
+const initialState: APIresponse = { articles: [], status: "", totalResults: 0 };
+
+export const fetchFromApi = createAsyncThunk("fetchFromApi", (url: string) =>
+  axios
+    .get(url)
+    .then((res) => res.data)
+    .catch()
+);
+
+const articlesSlice = createSlice({
+  name: "articles",
+  initialState,
+  reducers: {},
+  extraReducers: {
+    [fetchFromApi.pending.type]: (state, action) => {
+      state.status = "loading";
```

Comment:
maybe it would be better to call this part in the state 'isLoading' and let it be boolean.

Reply:
the state.status expects to receive a string.
maybe there is another way ?